### PR TITLE
Generate table of content for easier navigation in documentation pages

### PR DESCRIPTION
**Pull request by Tom32i**
- [x] Include the Content Table Processor as a part of Stenope
- [x] Generate table of content for easier navigation in documentation pages

Result :

| Desktop | Mobile |
| --- | --- |
| desktop | mobile |

### CODE REVIEW

**Review by ogizanagi — COMMENTED** [review]
Nice 👍🏻 

But now we moved this in the repository, we must cover it with unit and/or functional tests 🤓

**Comment by Tom32i** [comment]
Will do!

File: tests/Unit/TableOfContent/CrawlerTableOfContentGeneratorTest.php [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+        yield 'subset' => [
+            $content,
+            2,
+            3,
+            [
+                new Headline(2, 'AA', 'Suspendisse', [
+                    new Headline(3, 'AAA', 'Dolor'),
+                    new Headline(3, 'AAB', 'Sit amet'),
+                    new Headline(3, 'AAC', 'Magna non rhoncus', ),
```
Comment (ogizanagi) [inline tests/Unit/TableOfContent/CrawlerTableOfContentGeneratorTest.php:98]:
```suggestion
                    new Headline(3, 'AAC', 'Magna non rhoncus'),
```

File: src/Behaviour/TableOfContentGeneratorInterface.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+interface TableOfContentGeneratorInterface
+{
+    /**
+     * @param string   $content   An HTML DOM
+     * @param int|null $fromDepth Minimum title to include (default H1)
+     * @param int|null $toDepth   Maximum title to include (default H6)
+     *
+     * @return Headline[]
+     */
+    public function getTableOfContent(string $content, ?int $fromDepth = null, ?int $toDepth = null): array;
```
Comment (ogizanagi) [inline src/Behaviour/TableOfContentGeneratorInterface.php:20]: I've just some doubt on the usefulness of such an interface

**Review by ogizanagi — APPROVED** [review]

**Pull request merged**